src/dtclient/image.go

```diff
@@ -0,0 +1,121 @@
+package dtclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+type LatestImageInfo struct {
+	Source string `json:"source"`
+	Tag    string `json:"tag"`
+}
+
+func (dtc *dynatraceClient) GetLatestOneAgentImage() (*LatestImageInfo, error) {
+	latestImageInfo, err := dtc.processRequest(dtc.getLatestOneAgentImageUrl())
+
+	if err != nil {
+		log.Info("failed to process latest image response")
+		return nil, err
+	}
+
+	return latestImageInfo, nil
+}
+
+func (dtc *dynatraceClient) GetLatestCodeModulesImage() (*LatestImageInfo, error) {
+	latestImageInfo, err := dtc.processRequest(dtc.getLatestCodeModulesImageUrl())
+
+	if err != nil {
+		log.Info("failed to process latest image response")
+		return nil, err
+	}
+
+	return latestImageInfo, nil
+}
+
+func (dtc *dynatraceClient) GetLatestActiveGateImage() (*LatestImageInfo, error) {
+	latestImageInfo, err := dtc.processRequest(dtc.getLatestActiveGateImageUrl())
+
+	if err != nil {
+		log.Info("failed to process latest image response")
+		return nil, err
+	}
+
+	return latestImageInfo, nil
+}
+
+func (dtc *dynatraceClient) processRequest(url string) (*LatestImageInfo, error) {
```

Comment:
`processRequest` is a bit too generic, so maybe something like:
```suggestion
func (dtc *dynatraceClient) processLatestImageRequest(url string) (*LatestImageInfo, error) {
```